lib/router/transition-intent/named-transition-intent.js

```diff
@@ -114,7 +114,11 @@ export default class NamedTransitionIntent extends TransitionIntent {
           newHandlerInfo.context
         );
         var oldContext = oldHandlerInfo && oldHandlerInfo.context;
-        if (result.names.length > 0 && newHandlerInfo.context === oldContext) {
+        if (
+          result.names.length > 0 &&
+          !!oldContext &&
```

Comment:
We should be specific here I think. For example, a `0` or `’’` (empty string) are probably fine...